lib/SocketHelpers.cpp

```diff
@@ -147,9 +162,23 @@ std::optional<Message> ReceiveProxyWifiMessage(const wil::unique_socket& socket)
 
 static void SendBytes(wil::unique_socket& socket, gsl::span<const uint8_t> dataToSend)
 {
-    // TODO: Is it needed to loop on a send?
     while (dataToSend.size_bytes() > 0)
     {
+        // Wait until the destination is ready to receive
+        constexpr timeval timeout{.tv_sec{0}, .tv_usec{500000}}; // 0.5 sec timeout
+        fd_set write_set{};
+        FD_SET(socket.get(), &write_set);
+        const auto socket_ready = select(0, nullptr, &write_set, nullptr, &timeout);
+        if (socket_ready == SOCKET_ERROR)
+        {
+            THROW_LAST_ERROR_MSG("Error while waiting to send a message");
```

Comment:
Does `THROW_LAST_ERROR_MSG` call `WSAGetLastError()` to obtain the `WSA*` error value?

Reply:
`WSAGetLastError` is actually directly calling `GetLastError`, and we can safely assume the implementation won't change anymore.

If you feel strongly about it, I can change that, but we are already using it in this codebase (in code you wrote I think, I wonder if we didn't have the conversation with the opposite roles before! ^^)

Reply:
> If you feel strongly about it, I can change that, but we are already using it in this codebase (in code you wrote I think, I wonder if we didn't have the conversation with the opposite roles before! ^^)

You're probably right about this 😄 No strong feelings here, just wanted to make sure we report the correct error.